src/redux/tokens/reducer.ts

```diff
@@ -32,7 +32,24 @@ export default (state: ITokensConfigState = initialState, action: IAction) => {
                     }
                 };
             }
-
+        }
+        case UPDATE_TOKEN_CONTRACT_ADDRESS: {
+            return {
+                ...state,
+                [action.data.blockchain]: {
+                    ...state[action.data.blockchain],
+                    [action.data.chainId]: {
+                        ...state[action.data.blockchain][action.data.chainId],
```

Comment:
Make sure this is not going to crash.
For safety, we can use `...(state[action.data.blockchain] && state[action.data.blockchain][action.data.chainId])`